src/domain/model/test/interpretation/InterpretationRules.ts

```diff
@@ -0,0 +1,40 @@
+import { Results } from '../Results';
+import { Condition } from './Condition';
+import { OutputPattern } from './Output';
+import { Interpretation } from './Interpretation';
+import { DomainValidationError } from '../../DomainValidationError';
+
+export class InterpretationRules {
+  constructor(private rules: InterpretationRule[]) {}
+
+  interpret(results: Results): Array<Interpretation> {
+    return this.rules
+      .map((rule) => rule.interpret(results))
+      .filter((interpretation) => interpretation !== null) as Interpretation[];
+  }
+
+  static from(interpretationRulesSchema: Array<any>) {
+    if (!Array.isArray(interpretationRulesSchema)) {
+      throw new DomainValidationError('interpretationRules', 'InterpretationRules must be an array');
+    }
+    return new InterpretationRules(interpretationRulesSchema.map(InterpretationRule.from));
+  }
+}
+
+class InterpretationRule {
+  constructor(private output: OutputPattern, private condition: Condition) {}
+
+  interpret(results: Results): Interpretation | null {
+    if (this.condition.evaluate(results.details)) {
+      return this.output.evaluate(results);
+    }
+    return null;
+  }
+
+  static from(interpretationRuleSchema: any) {
+    return new InterpretationRule(
```

Comment:
Do you need validation in here, or is that handled somewhere else?

Reply:
Yeah.. could add. There will be an error thrown from the OutputPattern or the Condition based on what is invalid but better to have meaningful errors

Reply:
Added 👌